File: back/src/main/java/com/triptrace/domain/auth/auth/dto/SignupRequest.java
package com.triptrace.domain.auth.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public record SignupRequest(
    @NotBlank
    @Email
    String email,

    @NotBlank
    String username,

    @NotBlank
    String password,

    String profileImageUrl
) {
}

File: back/src/main/java/com/triptrace/domain/auth/auth/dto/SignupRequest.kt
package com.triptrace.domain.auth.auth.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

@JvmRecord
data class SignupRequest(
    @field:NotBlank
    @field:Email
    val email: String?,

    @field:NotBlank
    val username: String?,

    @field:NotBlank
    val password: String?,

    val profileImageUrl: String?
)
